### PR TITLE
[LOOP-2554] Detect delayed interaction with onboarding and reset to beginning

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-2554
- Fix bug with not restarting if prescription is claimed

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR for build: https://github.com/tidepool-org/LoopWorkspace/pull/490

File: TidepoolOnboarding/View Controllers/PrescriptionReviewUICoordinator.swift [hunk 1]
```diff
@@ -105,8 +111,10 @@ class PrescriptionReviewUICoordinator: UINavigationController, CompletionNotifyi
                 self?.setupCanceled()
             }
             prescriptionViewModel.didFinishStep = { [weak self] in
-                self?.therapySettingsViewModel = self?.constructTherapySettingsViewModel()
-                self?.stepFinished()
+                guard let self = self else { return }
+                self.prescriptionReviewDelegate?.prescriptionReview(hasNewPrescription: self.prescriptionViewModel.prescription!)
```
Comment (nhamming) [inline TidepoolOnboarding/View Controllers/PrescriptionReviewUICoordinator.swift:115]: why force unwrap here?  Is it not possible to pass in an optional to the function and have it decide what to do? Or should `self.prescriptionViewModel.prescription` be part of the `guard`?

**Review by nhamming — APPROVED** [review]
I don't want to block, but I do have a question about force unwrapping. If this is safe, could you include a comment explaining why?

File: TidepoolOnboarding/View Controllers/PrescriptionReviewUICoordinator.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolOnboarding/View Controllers/PrescriptionReviewUICoordinator.swift:115]: A prescription is 100% required by this point. If there is no prescription, then it is a complete logic fail and the app will not function correctly. I can add this to the guard statement, but if prescription does happen in real life, then they user will be stuck at a screen, tapping a button that does nothing.

Also, much of this has to be refactored a bit to better fit into onboarding. So, I'm not going to make the change at this point, but will consider this case when refactoring to hopefully prevent the forced unwrap in the first place.

**Pull request merged**